addons/account/models/account_tax.py

```diff
@@ -552,92 +552,782 @@ def onchange_price_include(self):
+                    'reverse': reverse,
+                }))
+                if rounding_method == 'round_per_line':
+                    tax_values['base'] = float_round(tax_values['base'], precision_rounding=prec_rounding)
+                    tax_values['display_base'] = float_round(tax_values['display_base'], precision_rounding=prec_rounding)
+
+        if skipped:
+            eval_tax_values_list = [tax_values for tax_values in eval_tax_values_list if tax_values['id'] not in skipped]
+
+        if eval_tax_values_list:
+            total_excluded = eval_tax_values_list[0]['base']
+            tax_amount = sum(tax_values['tax_amount_factorized'] for tax_values in eval_tax_values_list)
+            total_included = total_excluded + tax_amount
+        else:
+            total_included = total_excluded = evaluation_context['quantity'] * evaluation_context['price_unit']
+            if rounding_method == 'round_per_line':
+                total_included = total_excluded = float_round(
+                    total_excluded,
+                    precision_rounding=prec_rounding,
+                )
+
+        return {
+            'tax_values_list': eval_tax_values_list,
+            'total_excluded': total_excluded,
+            'total_included': total_included,
+            'price_include_added_base': price_include_added_base,
+        }
 
-        # We first need to find out whether this tax computation is made for a refund
-        tax_type = self and self[0].type_tax_use
-        is_refund = is_refund or (tax_type == 'sale' and price_unit > 0) or (tax_type == 'purchase' and price_unit < 0)
+    # -------------------------------------------------------------------------
+    # MAPPING PRICE_UNIT
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _adapt_price_unit_to_another_taxes(self, price_unit, original_tax_values_list, new_tax_values_list):
+        """ From the price unit and taxes given as parameter, compute a new price unit corresponding to the
+        new taxes.
 
-        rslt = self.with_context(caba_no_transition_account=True)\
-                   .compute_all(price_unit, currency=currency_id, quantity=quantity, product=product_id, partner=partner_id, is_refund=is_refund, include_caba_tags=include_caba_tags)
+        For example, from price_unit=106 and taxes=[6% tax-included], this method can compute a price_unit=121
+        if new_taxes=[21% tax-included].
 
-        return rslt
+        The price_unit is only adapted when all taxes in 'original_tax_values_list' are price-included even when
+        'new_tax_values_list' contains price-included taxes. This is made that way for the following example:
 
-    def flatten_taxes_hierarchy(self, create_map=False):
+        Suppose a fiscal position B2C mapping 15% tax-excluded => 6% tax-included.
+        If price_unit=100 with [15% tax-excluded], the price_unit is computed as 100 / 1.06 instead of becoming 106.
```

Comment:
@qdp-odoo As asked in the other PR, I added a comment explaining the weird behavior with fiscal position. I kept the previous behavior but I'm not able to justify why it should behave that way